File: 백준/Silver/2583. 영역 구하기/영역 구하기.py
m, n, k = [int(x) for x in input().split()]
matrix = [[True for _ in range(n)] for _ in range(m)]
vertices = [[int(x) for x in input().split()] for _ in range(k)]

for left_x, left_y, right_x, right_y in vertices:
    for y in range(left_y, right_y):
        for x in range(left_x, right_x):
            matrix[y][x] = False

count = 0
areas = []

def get_area(y:int, x:int) -> int:
    area = 0
    queue = [(y, x)]
    
    while queue:
        i, j = queue.pop(0)
        if i < 0 or j < 0 or i == m or j == n or matrix[i][j] is False: 
            continue
        area += 1
        matrix[i][j] = False
                    
        queue.append((i+1, j))
        queue.append((i-1, j))
        queue.append((i, j+1))
        queue.append((i, j-1))
    
    return area
    
for y in range(m):
    for x in range(n):
        if matrix[y][x]:
            count += 1
            areas.append(get_area(y, x))

print(count)
print(" ".join(map(str, sorted(areas))))

File: 백준/Silver/2583. 영역 구하기/영역 구하기.swift
let mnk = readLine()!.split(separator: " ").compactMap { Int($0) }
let (m, n, k) = (mnk[0], mnk[1], mnk[2])
let rectangles: [(leftY: Int, leftX: Int, rightY: Int, rightX:Int)] = (0..<k).map { _ in 
    let vertices = readLine()!.split(separator:" ").compactMap { Int($0) }
    return (vertices[1], vertices[0], vertices[3], vertices[2])
    }

var matrix = Array(repeating: Array(repeating: true, count: n), count: m)

for vertices in rectangles {
    for y in (vertices.leftY..<vertices.rightY) {
        for x in (vertices.leftX..<vertices.rightX) {
            matrix[y][x] = false
        }
    }
}

var count = 0
var areas = [Int]()

func getArea(_ i: Int, _ j: Int) -> Int {
    var area = 0
    var queue = [(y: i, x: j)]
    while !queue.isEmpty {
        let element = queue.removeFirst(), 
            y = element.y, 
            x = element.x
        
        if y < 0 || x < 0 || y >= m || x >= n || !matrix[y][x] {
            continue
        }
        matrix[y][x] = false
        
        area += 1
        
        queue.append((y+1, x))
        queue.append((y-1, x))
        queue.append((y, x+1))
        queue.append((y, x-1))
    }
    
    return area
}

(0..<m).forEach { i in 
    for j in (0..<n) where matrix[i][j] {
        count += 1
        
        areas.append(getArea(i, j))
    }
}
print(count)
print(areas.sorted(by: <).map { String($0) }.joined(separator: " "))